packages/ses/src/error/note-log-args.js

```diff
@@ -118,49 +123,69 @@ export const makeLRUCacheMap = budget => {
     return cell;
   };
 
+  /**
+   * @param {K} key
+   */
   const has = key => touchCell(key) !== undefined;
   freeze(has);
 
+  /**
+   * @param {K} key
+   */
   // TODO Change to the following line, once our tools don't choke on `?.`.
   // See https://github.com/endojs/endo/issues/1514
-  // const get = key => touchCell(key)?.value;
+  // const get = key => touchCell(key)?.data?.get(key);
   const get = key => {
     const cell = touchCell(key);
-    return cell && cell.value;
+    return cell && cell.data && cell.data.get(key);
   };
   freeze(get);
 
+  /**
+   * @param {K} key
+   * @param {V} value
+   */
   const set = (key, value) => {
     if (budget >= 1) {
+      /** @type {LRUCacheCell | undefined} */
       let cell = touchCell(key);
-      if (cell !== undefined) {
-        cell.value = value;
-      } else {
+      if (cell === undefined) {
         if (size >= budget) {
           const condemned = head.prev;
           spliceOut(condemned); // Drop least recently used
-          map.delete(condemned.key);
+          condemned.data = undefined;
           size -= 1;
         }
         size += 1;
-        cell = makeSelfCell(key, value);
-        map.set(key, cell);
+        cell = makeSelfCell(undefined);
         spliceAfter(head, cell); // start most recently used
       }
+      if (!cell.data) {
+        // Either a new cell or a condemned cell.  Add its data.
+        cell.data = new WeakMap();
+        // Advertise the cell for this key.
+        keyToCell.set(key, cell);
+      }
+      // Update the data.
+      cell.data.set(key, value);
     }
     // eslint-disable-next-line no-use-before-define
     return lruCacheMap; // Needed to be WeakMap-like
   };
   freeze(set);
 
   // "delete" is a keyword.
+  /**
+   * @param {K} key
+   */
   const deleteIt = key => {
-    const cell = map.get(key);
+    const cell = keyToCell.get(key);
     if (cell === undefined) {
       return false;
     }
     spliceOut(cell);
-    map.delete(key);
+    cell.data = undefined;
+    keyToCell.delete(key);
     size -= 1;
```

Comment:
if `cell.data` was already undefined, we should not decrement the size as it was already condemned.